crates/bevy_ecs/src/reflect.rs

```diff
@@ -27,7 +27,36 @@ pub struct ReflectComponent {
     copy: fn(&World, &mut World, Entity, Entity),
 }
 
+/// The raw functions needed to make up a [`ReflectComponent`].
+/// 
+/// This is used when creating custom implementations of [`ReflectComponent`] with
+/// [`ReflectComponent::new()`].
+pub struct ReflectComponentFns {
+    pub insert: fn(&mut World, Entity, &dyn Reflect),
```

Comment:
Could you add a doc comment for each of these (same for `ReflectResourceFns`)? I imagine the people who do need this could benefit from knowing what they actually do.